crapsgame/crapsgame.js

```diff
@@ -0,0 +1,207 @@
+// Craps Main Data
+let crapsUsername = ""
+
+// Craps Game Settings
+const startingMoney = 1000
+const startingRounds = 0
+const bets = {
+	even: "EVEN",
+	odd: "ODD"
+}
+const minimumBet = 100
+
+// Craps Dice Roll Settings
+const numDiceToRoll = 2
+const hideDiceDelayMs = 10000000
+const processDiceResultDelayMs = 1800
+
+// HTML Element IDs
+const crapsUsernameInput = "craps-username-input"
+const crapsRegistrationPane = "craps-registration-pane"
+const crapsMainSection = "craps-main-section"
+const crapsStatsUsername = "craps-stats-username"
+const crapsStatsMoney = "craps-stats-money"
+const crapsStatsRounds = "craps-stats-rounds"
+const crapsUserBetAmount = "craps-user-bet-amount"
+const crapsRollDiceButton = "craps-roll-dice-button"
+const crapsRollDiceAnimationContainer = "craps-roll-dice-animation-container"
+const crapsBettingGridContainer = "craps-betting-grid-container"
+const crapsRoundFinishGridContainer = "craps-round-finish-grid-container"
+const crapsRoundFinishMessage = "craps-round-finish-message"
+const crapsNextRoundButton = "craps-next-round-button"
+const crapsNextRoundButtonDisabled = "craps-next-round-button-disabled"
+
+// In-game variables
+let currentMoney = startingMoney
+let currentRounds = startingRounds
+let currentBet = bets.even
+let currentBetAmount = minimumBet
+let canChangeBet = true
+
+// HTML ELEMENT Manipulation Functions
+
+function showElement (elementId) {
+	document.getElementById(elementId).style.display = "block"
+}
+
+function hideElement (elementId) {
+	document.getElementById(elementId).style.display = "none"
+}
+
+function showRegistrationPane () {
+	showElement(crapsRegistrationPane)
+}
+
+function removeRegistrationPane () {
+	hideElement(crapsRegistrationPane)
+}
+
+function showMainGameSection () {
+	showElement(crapsMainSection)
+}
+
+function hideMainGameSection () {
+	hideElement(crapsMainSection)
+}
+
+// Game Starting Point
+
+function registerCrapsPlayer () {
+	crapsUsername = document.getElementById(crapsUsernameInput).value
+
+	// Username validation check
+	let firstCharIsDigitRegex = /^[0-9]|[^a-zA-Z0-9_]/g
```

Comment:
What is this regex checking?

Reply:
Making sure the username contains only alphanum characters or underscore, and no spaces